OAuth2/AuthorizationRoot.cs

```diff
@@ -69,7 +69,8 @@ public virtual IEnumerable<IClient> Clients
         /// </summary>        
         protected virtual IEnumerable<Type> GetClientTypes()
         {
-            return Assembly.GetExecutingAssembly().GetTypes().Where(typeof (IClient).IsAssignableFrom);
+          var clientType = typeof(IClient);
+          return AppDomain.CurrentDomain.GetAssemblies().SelectMany(s => s.GetTypes()).Where(p => clientType.IsAssignableFrom(p));
```

Comment:
Agree regarding the clientType variable, do I need to amend and send a new PR? With regards to performance I've not tested in any projects that have 00's or 000's of assemblies - I have no projects of that sort of size but if performance is a concern we could consider adding a configuration option somewhere telling it where to look for them?